frontend/pages/set-new-password.tsx

```diff
@@ -0,0 +1,162 @@
+      newPassword2: $newPasswordRepeat
+    ) {
+      success,
+      errors
+    }
+  }
+`
+
+const validationSchema = yup.object({
+  resetToken: yup.string().required(),
+  newPassword: yup.string()
+  .min(8, 'Password must be at least 8 characters')
+  .required('Password is required'),
+  newPasswordRepeat: yup.string()
+  .min(8, 'Password must be at least 8 characters')
+  .oneOf([yup.ref('newPassword'), null], 'Passwords must match')
+  .required('Please re-enter your password')
+})
+
+
+function SetNewPasswordForm() {
+
+  const [errorState, setErrorState] = useState(false)
+
+  const [resetPassword] = useMutation(PASSWORD_RESET, {
+      onCompleted(data){
+        if(data.passwordReset.success === true){
+          router.push("/sign-in")
+        }
+        else{
+          setErrorState(true)
+        }
+      },
+  })
+
+  const resetPasswordForm = useFormik({
+    initialValues: {
+        resetToken: '',
+        newPassword: '',
+        newPasswordRepeat: ''
+      },
+      validationSchema: validationSchema,
+      onSubmit: (values) => {
+        resetPassword({
+          variables:
+          {
+            ...values
+          }
+        })
+      }
+  })
+
+  const handleDialogClose = () => {
+    resetPasswordForm.resetForm();
+    setErrorState(false)
+  }
+
+
+  return (
+    <React.Fragment>
+      <AppAppBar />
+      <PageContainer title="Set New Password">
+      <div style={{ padding: "48px 16px" }}>
+        <Container maxWidth="xs">
+                  <form onSubmit={resetPasswordForm.handleSubmit}>
+                  <TextField
+                  id="resetToken"
+                  label="Reset Token"
+                  variant="outlined"
+                  fullWidth
+                  required
+                  value={resetPasswordForm.values.resetToken}
+                  onChange={resetPasswordForm.handleChange}
+                  InputLabelProps = {{
+                    shrink: true
+                  }}
+                  style = {{ margin: 8}}
+                  />
+                  <TextField
+                    id="newPassword"
+                    label="New Password"
+                    variant="outlined"
+                    type="password"
+                    fullWidth
+                    required
+                    value={resetPasswordForm.values.newPassword}
+                    onChange={resetPasswordForm.handleChange}
+                    InputLabelProps = {{
+                      shrink: true
+                    }}
+                    style = {{ margin: 8}}
+                  />
+                  <TextField 
+                    id="newPasswordRepeat"
+                    label="Repeat New Password"
+                    variant="outlined"
+                    fullWidth
+                    required
+                    type="password"
+                    value={resetPasswordForm.values.newPasswordRepeat}
+                    onChange={resetPasswordForm.handleChange}
```

Comment:
These text fields don't have the error attribute, so they won't display the formik errors for min number of chars and same password

Reply:
added